x-pack/plugins/alerting/server/lib/trim_recovered_alerts.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Logger } from '@kbn/logging';
+import { map } from 'lodash';
+import { Alert } from '../alert';
+import { AlertInstanceState, AlertInstanceContext } from '../types';
+
+interface TrimmedRecoveredAlertsResult<
+  State extends AlertInstanceState,
+  Context extends AlertInstanceContext,
+  RecoveryActionGroupIds extends string
+> {
+  trimmedAlertsRecovered: Record<string, Alert<State, Context, RecoveryActionGroupIds>>;
+  trimmedAlertsRecoveredCurrent: Record<string, Alert<State, Context, RecoveryActionGroupIds>>;
+}
+
+export interface TrimRecoveredOpts {
+  index: number | string;
```

Comment:
Is it typed this way because of the lodash `map()` call?  I'd prefer to make this more specific (number), perhaps do the cast in the `map()` iterator, so that we don't have to `as number` elsewhere.

Reply:
This was number and string because the rule registry uses arrays and the framework uses Records. It should be a string now. Resolved in this commit 0ad43e7cc1145597b7b3743a42ef781003dc4133